lua/packer.lua

```diff
@@ -369,11 +369,40 @@ packer.sync = function(...)
   end)()
 end
 
+
+local function reload_module(name)
+  if name then
+    package.loaded[name] = nil
+    return require(name)
+  end
+end
+
+--- Search through all the loaded packages for those that
+--- return a function, then cross reference them with all
+--- the plugin configs and setups and if there are any matches
+--- reload the user module.
+local function precompile(plugs)
+  local reverse_index = {}
+  for k, v in pairs(package.loaded) do
```

Comment:
We might be able to replace this loop with `vim.tbl_add_reverse_lookup`. Not a big deal either way - this loop is certainly more specialized to our needs.

Reply:
Tbh I couldn't
 a) tell if it would mutate the table in place, and given it's the `package.loaded` table I didn't want to risk it
 b) it works by adding the reverse look up to the existing table (copy or not is unclear), seems like it'd double the size of this table which isn't a huge deal since we aren't iterating through it, but makes it harder to make sense of 🤷🏿‍♂️ 

TLDR: it just felt like it'd only be a slight improvement/if that over just a loop that does specifically whats needed.

Reply:
Totally reasonable; I'm good with this tradeoff.